From e7e3b0f956929f022caa01ed25a482495b11d575 Mon Sep 17 00:00:00 2001
From: xuchenghua <xuchenghua@loongson.cn>
Date: Tue, 8 Dec 2015 19:44:43 +0800
Subject: [PATCH 03/15] Added misses sync in mips_process_sync_loop for add
 sync before ll sc jump label.

    mips/
    mips.c:  mips_process_sync_loop(): Added sync.
---
 gcc/config/mips/mips.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gcc/config/mips/mips.c b/gcc/config/mips/mips.c
index 172134af347..0cc99a6c86b 100644
--- a/gcc/config/mips/mips.c
+++ b/gcc/config/mips/mips.c
@@ -14118,6 +14118,9 @@ mips_process_sync_loop (rtx_insn *insn, rtx *operands)
   /* Output the branch-back label.  */
   mips_multi_add_label ("1:");
 
+  /* loongson needed */
+  mips_multi_add_insn ("sync", NULL);
+
   /* OLDVAL = *MEM.  */
   mips_multi_add_insn (is_64bit_p ? "lld\t%0,%1" : "ll\t%0,%1",
 		       oldval, mem, NULL);
@@ -14234,6 +14237,7 @@ mips_process_sync_loop (rtx_insn *insn, rtx *operands)
   if (required_oldval)
     {
       mips_multi_add_label ("2:");
+      /* loongson needed. */
       mips_multi_add_insn ("sync", NULL);
     }
 
-- 
2.16.1

